x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/exploration_page_wrapper/exploration_page_wrapper.tsx

```diff
@@ -33,24 +33,32 @@ import { FeatureImportanceSummaryPanelProps } from '../total_feature_importance_
 import { useExplorationUrlState } from '../../hooks/use_exploration_url_state';
 import { ExplorationQueryBarProps } from '../exploration_query_bar/exploration_query_bar';
 
-const filters = {
-  options: [
-    {
-      id: 'training',
-      label: i18n.translate('xpack.ml.dataframe.analytics.explorationResults.trainingSubsetLabel', {
-        defaultMessage: 'Training',
-      }),
-    },
-    {
-      id: 'testing',
-      label: i18n.translate('xpack.ml.dataframe.analytics.explorationResults.testingSubsetLabel', {
-        defaultMessage: 'Testing',
-      }),
-    },
-  ],
-  columnId: 'ml.is_training',
-  key: { training: true, testing: false },
-};
+function getFilters(resultsField: string) {
+  return {
+    options: [
+      {
+        id: 'training',
+        label: i18n.translate(
+          'xpack.ml.dataframe.analytics.explorationResults.trainingSubsetLabel',
+          {
+            defaultMessage: 'Training',
+          }
+        ),
+      },
+      {
+        id: 'testing',
+        label: i18n.translate(
+          'xpack.ml.dataframe.analytics.explorationResults.testingSubsetLabel',
+          {
+            defaultMessage: 'Testing',
+          }
+        ),
+      },
+    ],
+    columnId: `${resultsField}.is_training`,
```

Comment:
do we need to apply an encoding in case the `resultsFields` contains special characters?